libs/api-provider-environment/src/transaction/useTransactionClientStorage.tsx

```diff
@@ -5,18 +5,25 @@ import transactionDB from './client-storage/db';
 export type UseTxClientStorageReturnType = {
   transactions: TransactionType[];
   addNewTransaction: (tx: TransactionType) => void;
+  addTransactions: (txs: TransactionType[]) => void;
   clearTxHistory: () => void;
 };
 
 const useTxClientStorage = () => {
   const transactions = useLiveQuery(() => transactionDB.txItems.toArray());
 
-  const addNewTransaction = (tx: TransactionType) => {
+  const addNewTransaction = async (tx: TransactionType) => {
+    // if the tx already exists, an error will be thrown
     transactionDB.txItems.add({ ...tx });
   };
 
-  const clearTxHistory = () => {
-    transactionDB.transaction('rw', transactionDB.txItems, async () => {
+  const addTransactions = async (txs: TransactionType[]) => {
+    // if any tx already exist, an error will be thrown
```

Comment:
Same here, did we handle the error that would be thrown here?

Reply:
Errors will be handled when we call the function on other files.Ï